### PR TITLE
Specify table name for TableBasedAuthority

### LINKED ISSUE #113: TableBasedAuthority looking for the wrong index
I think this line needs to change:
https://github.com/projecthydra-labs/questioning_authority/blob/master/lib/qa/authorities/local/table_based_authority.rb#L6
to `!conn.indexes('qa_local_authority_entries')`

I'm seeing errors because it's looking for the `local_authority_entries` table, which doesn't exist.

### PR DESCRIPTION
**Pull request by awead**
Fixes #113

### CODE REVIEW

File: lib/qa/authorities/local/table_based_authority.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+    self.table_name = "qa_local_authority_entries"
+
     def self.check_for_index
       @checked_for_index ||= begin
         conn = ActiveRecord::Base.connection
-        if table_or_view_exists? && !conn.indexes('local_authority_entries').find { |i| i.name == 'index_local_authority_entries_on_lower_label' }
+        if table_or_view_exists? && !conn.indexes(table_name).find { |i| i.name == 'index_local_authority_entries_on_lower_label' }
           Rails.logger.error "You've installed local authority tables, but you haven't indexed the label.  Rails doesn't support functional indexes in migrations, so you'll have to add this manually:\n" \
-            "CREATE INDEX \"index_qa_local_authority_entries_on_lower_label\" ON \"qa_local_authority_entries\" (local_authority_id, lower(label))\n" \
+            "CREATE INDEX \"index_#{table_name}_on_lower_label\" ON \"#{table_name}\" (local_authority_id, lower(label))\n" \
```
Comment (jcoyne) [inline lib/qa/authorities/local/table_based_authority.rb]: perhaps you should extract an `index_name` method?

**Comment by awead** [comment]
@jcoyne ok, done and added the same options to the MysqlTableBasedAuthority

File: lib/qa/authorities/local/mysql_table_based_authority.rb [hunk 2]
```diff
@@ -1,15 +1,19 @@
 module Qa
   module Authorities
     module Local
-      #
       class MysqlTableBasedAuthority < Local::TableBasedAuthority
+
+        self.table_name = "qa_local_authority_entries"
```
Comment (jcoyne) [inline lib/qa/authorities/local/mysql_table_based_authority.rb]: This line is unnecessary, as it's inherited.

File: spec/lib/authorities/mysql_table_based_authority_spec.rb [hunk 3]
```diff
@@ -8,6 +8,16 @@
     Qa::Authorities::Local.register_subauthority('language', described_class.to_s)
   end
 
+  describe "::table_name" do
```
Comment (jcoyne) [inline spec/lib/authorities/mysql_table_based_authority_spec.rb]: why the preceding `::`?

File: lib/qa/authorities/local/mysql_table_based_authority.rb (hunk 2 above)
Comment (awead) [inline lib/qa/authorities/local/mysql_table_based_authority.rb]: duh!

File: spec/lib/authorities/mysql_table_based_authority_spec.rb (hunk 3 above)
Comment (awead) [inline spec/lib/authorities/mysql_table_based_authority_spec.rb]: Shorthand for class method? Instance methods are proceeded by #, or at least that's what I see in the Ruby docs.

**Comment by awead** [comment]
@jcoyne alright, once more for the west coast...

**Pull request merged**